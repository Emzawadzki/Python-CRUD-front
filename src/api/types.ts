export interface ErrorResponse {
  data: null;
  error: string; // TODO: add errors enum/typing
}
export interface SuccessResponse<T> {
  data: T;
  error: null;
}

export type ApiResponse<T> = SuccessResponse<T> | ErrorResponse;

export interface RawPerson {
  id: number;
  first_name: string;
  last_name: string;
}

export interface Person {
  id: RawPerson["id"];
  firstName: RawPerson["first_name"];
  lastName: RawPerson["last_name"];
}
