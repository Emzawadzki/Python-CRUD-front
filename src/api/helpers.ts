import { RawPerson, Person } from "./types";

export const parsePerson = (person: RawPerson): Person => ({
  id: person.id,
  firstName: person.first_name,
  lastName: person.last_name,
});
