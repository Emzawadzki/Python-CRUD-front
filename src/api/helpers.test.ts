import { parsePerson } from "./helpers";
import { RawPerson } from "./types";

const testPerson: RawPerson = {
  id: 6,
  first_name: "test-first-name",
  last_name: "test-last-name",
};

describe("persePerson", () => {
  it("should return parsed data", () => {
    expect(parsePerson(testPerson)).toMatchSnapshot();
  });
});
