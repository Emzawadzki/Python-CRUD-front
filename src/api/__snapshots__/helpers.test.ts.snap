// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`persePerson should return parsed data 1`] = `
Object {
  "firstName": "test-first-name",
  "id": 6,
  "lastName": "test-last-name",
}
`;
